### PR TITLE
Better error for any_of()

### LINKED ISSUE #241: add example of probable source of error to error message
Just a very minor proposal to add to the error message at https://github.com/r-lib/rlang/blob/1a53bd085c8e0ed41dcc56cce334dde926da7067/R/dots-ellipsis.R#L169

"These dots only exist to allow future extensions and should be empty."
"Did you misspecify an argument?"

It seems a common way to reach this error is forgetting to use c() e.g. I was using any_of("a","b","c") when it should have been any_of(c("a","b","c")) 

Would it be acceptable to add "Did you misspecify an argument? e.g. forget to use c()"?" to this error message? I think it could be helpful for beginners.

### PR DESCRIPTION
**Pull request by hadley**
Fixes #241

### CODE REVIEW

File: R/helpers-vector.R [hunk 1]
```diff
@@ -88,6 +88,12 @@ all_of <- function(x) {
 #' @export
 any_of <- function(x, ..., vars = NULL) {
   vars <- vars %||% peek_vars(fn = "any_of")
-  check_dots_empty()
+  if (!missing(...)) {
+    abort(c(
```
Comment (lionel-) [inline R/helpers-vector.R:92]: Use `cli_abort()` with `.code` style?

**Review by lionel- — APPROVED** [review]

File: R/helpers-vector.R (hunk 1 above)
Comment (hadley) [reply R/helpers-vector.R:92]: How about I convert them all in a separate PR?

Comment (hadley) [reply R/helpers-vector.R:92]: I see you filed #285 so we're thinking the same way 😄

Comment (lionel-) [reply R/helpers-vector.R:92]: Up to you. Personally for this sort of things I generally convert any code I touch ahead of time, before the bulk changes. Especially for this sort of dull but tricky changes that are so easy to mess up when doing a mass update.

Comment (hadley) [reply R/helpers-vector.R:92]: Yeah, good point.

**Pull request merged**